src/components/refactor/dialogs/generator/set-points/frequency-regulation.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    gridItem,
+    gridItemWithTooltip,
+    percentageTextField,
+} from '../../../../dialogs/dialogUtils';
+import { useWatch } from 'react-hook-form';
+import { DROOP, FREQUENCY_REGULATION } from '../../../utils/field-constants';
+import React from 'react';
+import FloatInput from '../../../rhf-inputs/float-input';
+import SwitchInput from '../../../rhf-inputs/booleans/switch-input';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import CheckboxNullableInput from 'components/refactor/rhf-inputs/boolean-nullable-input';
+import { Box } from '@mui/material';
+
+const FrequencyRegulation = ({ isGeneratorModification, previousValues }) => {
+    const watchFrequencyRegulation = useWatch({
+        name: FREQUENCY_REGULATION,
+    });
+
+    const isFrequencyRegulationOn =
+        watchFrequencyRegulation === true ||
+        (watchFrequencyRegulation === null &&
+            previousValues?.activePowerControlOn === true);
+
```

Comment:
add watcher for previousValue: 
const watchFrequencyRegulationPreviousValue = useWatch({
        name: PREVIOUS_FREQUENCY_REGULATION
    })

     const isFrequencyRegulationOn =
         watchFrequencyRegulation === true ||
         (watchFrequencyRegulation === null &&
            watchFrequencyRegulationPreviousValue);